storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,604 @@
+        Map<Integer, String> taskIdToComponentId = new HashMap<Integer, String>();
+
+        StormTopology systemTopology = systemTopology(stormConf, userTopology);
+        Map<String, Object> components = allComponents(systemTopology);
+        Map<String, Integer> componentIdToTaskNum = new TreeMap<String, Integer>();
+        for (Map.Entry<String, Object> entry : components.entrySet()) {
+            Map conf = componentConf(entry.getValue());
+            Object taskNum = conf.get(Config.TOPOLOGY_TASKS);
+            if (taskNum != null) {
+                componentIdToTaskNum.put(entry.getKey(), Utils.getInt(taskNum));
+            }
+        }
+
+        int taskId = 1;
+        for (Map.Entry<String, Integer> entry : componentIdToTaskNum.entrySet()) {
+            String componentId = entry.getKey();
+            Integer taskNum = entry.getValue();
+            while (taskNum > 0) {
+                taskIdToComponentId.put(taskId, componentId);
+                taskNum--;
+                taskId++;
+            }
+        }
+        return taskIdToComponentId;
+    }
+
+    public static List<Integer> executorIdToTasks(List<Long> executorId) {
+        List<Integer> taskIds = new ArrayList<Integer>();
+        int taskId = executorId.get(0).intValue();
+        while (taskId <= executorId.get(1).intValue()) {
+            taskIds.add(taskId);
+            taskId++;
+        }
+        return taskIds;
+    }
+
+    public static Map<Integer, NodeInfo> taskToNodeport(Map<List<Long>, NodeInfo> executorToNodeport) {
+        Map<Integer, NodeInfo> tasksToNodeport = new HashMap<Integer, NodeInfo>();
+        for (Map.Entry<List<Long>, NodeInfo> entry : executorToNodeport.entrySet()) {
+            List<Integer> taskIds = executorIdToTasks(entry.getKey());
+            for (Integer taskId : taskIds) {
+                tasksToNodeport.put(taskId, entry.getValue());
+            }
+        }
+        return tasksToNodeport;
+    }
+
+    public static IAuthorizer mkAuthorizationHandler(String klassName, Map conf) {
+        return _instance.mkAuthorizationHandlerImpl(klassName, conf);
+    }
+
+    protected IAuthorizer mkAuthorizationHandlerImpl(String klassName, Map conf) {
+        IAuthorizer aznHandler = null;
+        try {
+            if (klassName != null) {
+                Class aznClass = Class.forName(klassName);
+                if (aznClass != null) {
+                    aznHandler = (IAuthorizer) aznClass.newInstance();
+                    if (aznHandler != null) {
+                        aznHandler.prepare(conf);
+                    }
+                    LOG.debug("authorization class name:{}, class:{}, handler:{}",klassName, aznClass, aznHandler);
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Failed to make authorization handler, klassName:{}", klassName);
+        }
```

Comment:
The original code did not catch any exceptions it just threw them.  We should do the same.